### ISSUE TITLE
Analysis on GEOGLAM Crop Conditions fails with "Network Error"

### ISSUE THREAD

**Issue by j08lue** [NASA-IMPACT/veda-data-pipelines#350]
Steps to reproduce:
1. Go to https://www.earthdata.nasa.gov/dashboard (v 0.8.0)
3. Choose "Analysis" tab
4. Draw area somewhere over land and enter range 2020-01-02 to 2023-01-01
5. Select GEOGLAM Crop Conditions dataset
6. Choose generate

Or use this link:
https://www.earthdata.nasa.gov/dashboard/analysis/results?start=2020-01-01T00%3A00%3A00.000Z&end=2023-01-01T00%3A00%3A00.000Z&datasetsLayers=geoglam&aoi=%7CenB%7DhhzDc%7BhyB%3FftcAlxo_BzedwBureE

<img src="https://github.com/NASA-IMPACT/veda-config/assets/3404817/ba20e35e-9f35-41c3-9565-00890ebbe889">

**Comment by smohiudd** [comment]
Looks like a CORS error for the `/statistics` endpoint in the raster API. I can look into this. 

<img width="1222" alt="image" src="https://github.com/NASA-IMPACT/veda-data-pipelines/assets/34844565/750f0dd4-1228-48ab-ad4c-0c4dbe5c288c">

**Comment by j08lue** [comment]
Yeah, CORS after 500 is a classic red herring, right?

**Comment by vincentsarago** [comment]
> At least one of the problem COGs seem to be valid but use a different compression and lack overviews (LZW)
i.e. here is a valid COG from the error log above
https://staging-raster.delta-backend.com/cog/validate?url=s3://veda-data-store-staging/geoglam/CropMonitor_202109.tif

> But a newer COG that uses DEFLATE is working with the stats endpoint https://staging-raster.delta-backend.com/cog/statistics?url=s3://veda-data-store-staging/geoglam/CropMonitor_2023_03_28.tif.

>I don't think the compression type is the issue, just a clue that some of these files were created differently than others.

The first file doesn't have overviews so to get the `statistics` rio-tiler will fetch the whole data and I guess the request will timeout! 

You can see that for the second file there are plenty overviews and the statistics method return fast

**Comment by anayeaye** [comment]
Thanks @vincentsarago that makes sense, 

So the resolution here is to reprocess the COGs without overviews. There are only 36 tiffs altogether.

**Comment by smohiudd** [comment]
Thanks @vincentsarago. I'll work on reprocessing these geoglam COGs.

**Comment by j08lue** [comment]
Sorry to hear. Please post the command you use here to fix the files, then I will be sure to apply it to all incoming files.

**Comment by smohiudd** [comment]
That 500 could be because there is no data included within that bbox over Africa. I tried this POST on the same cog and it returned:

```
curl 'https://staging-raster.delta-backend.com/cog/statistics?url=s3://veda-data-store-staging/geoglam/CropMonitor_202208.tif' -X POST -H 'Content-Type: application/json' --data-raw '{
      "type": "Feature",
      "properties": {},
      "geometry": {
        "coordinates": [
          [
            [
              23.759006417039927,
              52.36779621746979
            ],
            [
              23.759006417039927,
              47.976621171066455
            ],
            [
              35.17015504381041,
              47.976621171066455
            ],
            [
              35.17015504381041,
              52.36779621746979
            ],
            [
              23.759006417039927,
              52.36779621746979
            ]
          ]
        ],
        "type": "Polygon"
      }
    }'
```
`{"type":"Feature","geometry":{"coordinates":[[[23.759006417039927,52.36779621746979],[23.759006417039927,47.976621171066455],[35.17015504381041,47.976621171066455],[35.17015504381041,52.36779621746979],[23.759006417039927,52.36779621746979]]],"type":"Polygon"},"properties":{"statistics":{"1":{"min":3.0,"max":7.0,"mean":3.4466395379364663,"count":243776.0,"sum":840208.0,"std":0.966422116925274,"median":3.0,"majority":3.0,"minority":4.0,"unique":4.0,"histogram":[[196243.0,0.0,1098.0,0.0,0.0,38979.0,0.0,0.0,0.0,7456.0],[3.0,3.4,3.8,4.2,4.6,5.0,5.4,5.800000000000001,6.2,6.6,7.0]],"valid_percent":60.42,"masked_pixels":159680.0,"valid_pixels":243776.0,"percentile_2":3.0,"percentile_98":7.0}}}`

**Comment by danielfdsilva** [comment]
oooh. I think you're right.
<img width="846" alt="image" src="https://github.com/NASA-IMPACT/veda-data-pipelines/assets/1090606/57d8b376-a049-4a6e-9736-28d3dcc7d826">

Then, can we have an error message that it is not a 500? That would make it easier to handle and differentiate from other errors.

**Comment by smohiudd** [comment]
Agreed, that 500 error is really confusing. I'll see what we can do to change it.

**Comment by smohiudd** [comment]
@vincentsarago what you think about have a 404 options for statistics in titiler: https://github.com/developmentseed/titiler/blob/921ca0fac204128755430180f7a1526eb181b3e5/src/titiler/core/titiler/core/factory.py#L418-L477

```
responses={
                404: {"model": Message, "description": "Statistics not found"},
                200: {
                    "content": {"application/json": {}},
                    "description": "Return dataset's statistics.",
                }
            }
```

**Comment by vincentsarago** [comment]
@smohiudd I'm not sure to understand can you summarize the issue?

**Comment by smohiudd** [comment]
The issue is if you use `/cog/statistics` POST with an area that has no data, then we're getting this error in the lambda which results in a 500 error:


```
...
k2d0vzid/lib/python/titiler/core/factory.py", line 411, in geojson_statistics 
File "/tmp/pip-target-k2d0vzid/lib/python/rio_tiler/utils.py", line 166, in get_array_statistics 
File "<__array_function__ internals>", line 200, in percentile 
File "/tmp/pip-target-k2d0vzid/lib/python/numpy/lib/function_base.py", line 4205, in percentile 
File "/tmp/pip-target-k2d0vzid/lib/python/numpy/lib/function_base.py", line 4473, in _quantile_unchecked 
File "/tmp/pip-target-k2d0vzid/lib/python/numpy/lib/function_base.py", line 3752, in _ureduce 
File "/tmp/pip-target-k2d0vzid/lib/python/numpy/lib/function_base.py", line 4639, in _quantile_ureduce_func 
File "/tmp/pip-target-k2d0vzid/lib/python/numpy/lib/function_base.py", line 4750, in _quantile 
File "<__array_function__ internals>", line 200, in take 
File "/tmp/pip-target-k2d0vzid/lib/python/numpy/core/fromnumeric.py", line 190, in take
 File "/tmp/pip-target-k2d0vzid/lib/python/numpy/core/fromnumeric.py", line 57, in _wrapfunc 

IndexError: cannot do a non-empty take from an empty axes.
```

What about a conditional to check if the area has valid data if not then return 404

**Comment by vincentsarago** [comment]
that should be fixed in rio-tiler directly. DO you know the version of titiler and rio-tiler you are using?

**Comment by vincentsarago** [comment]
I'll fix this in rio-tiler tomorrow 

```python
import numpy
from rio_tiler.utils import get_array_statistics

arr = numpy.ma.array(numpy.zeros((1,256,256)))
arr.mask = True
get_array_statistics(arr)

IndexError: cannot do a non-empty take from an empty axes
```

**Comment by smohiudd** [comment]
I see titler `titiler.application>=0.10,<0.11` in the veda-back raster requirements but no rio-tiler

**Comment by smohiudd** [comment]
Thanks @vincentsarago

**Comment by danielfdsilva** [comment]
@smohiudd Is there anything left on the backend side, or did this address the issue?

**Comment by smohiudd** [comment]
We just need to re deploy the cloud formation stack so its picks up the new rio-tiler version that deals with the no data pixels. That should take care of the 500 error we were seeing before.

**Comment by smohiudd** [comment]
Isn't this it? https://pypi.org/project/rio-tiler/4.1.12/

**Comment by jsignell** [comment]
oh! sorry for the noise. I didn't realize it was an patch release on an old version. Ignore me.

**Comment by smohiudd** [comment]
No problem !

**Comment by j08lue** [comment]
I can confirm the analysis task that before was failing now works - at least on the attempt I just made, using the direct link mentioned [above](https://github.com/NASA-IMPACT/veda-data-pipelines/issues/350#issue-1738017075) and some modifications.

Great team work!

**Comment by smohiudd** [comment]
@danielfdsilva does this work ok in `dev` but not `staging`?
